### PR TITLE
[ASCollectionView] When Resizing, Invalidate Layout After Remeasuring

### PR DESCRIPTION
**Pull request by Adlai-Holler**
This fixes rotation issues @rcancro and I have seen. It is still being tested.

### CODE REVIEW

**Comment by Adlai-Holler** [comment]
OK this is working super good! Could this get some inspection from @maicki @levi @appleguy

File: AsyncDisplayKit/ASCollectionView.mm [hunk 1]
```diff
@@ -857,7 +857,7 @@ - (void)layoutSubviews
   _nextLayoutInvalidationStyle = ASCollectionViewInvalidationStyleNone;
   switch (invalidationStyle) {
     case ASCollectionViewInvalidationStyleWithAnimation:
-      if (!_superPerformingBatchUpdates) {
+      if (0 == _superBatchUpdateCount) {
```
Comment (maicki) [inline AsyncDisplayKit/ASCollectionView.mm]: @appleguy @Adlai-Holler What is the opinion about the code style? I personally would prefer `0 == _superBatchUpdateCount`, but we should find a way that works for all and the future

**Comment by maicki** [comment]
One comment but otherwise looks good

**Comment by hannahmbanana** [comment]
Assuming this passes user testing, it sounds like this should be cherry picked to 6.8

File: AsyncDisplayKit/ASCollectionView.mm (hunk 1 above)
Comment (Adlai-Holler) [inline AsyncDisplayKit/ASCollectionView.mm]: For BOOLs I prefer `NO == _myBool` so you can immediately see the NO. For all other assignments this "defensive ordering" with constant on the left prevents people from accidentally using a single =, so I think we should standardize on that.

Comment (rcancro) [inline AsyncDisplayKit/ASCollectionView.mm]: FWIW, I'm pretty sure the compiler catches a single = for you now. 
screen shot 2016-07-15 at 4 19 30 pm

Comment (Adlai-Holler) [inline AsyncDisplayKit/ASCollectionView.mm]: Oh neat!

Comment (appleguy) [inline AsyncDisplayKit/ASCollectionView.mm]: @Adlai-Holler I understand the value in this style, but for now let's match the prevailing style, which is == NO

**Pull request merged**